sample-app/tests/browser/pageoperator.js

```diff
@@ -0,0 +1,58 @@
+const { BrowserPageWidths } = require("./constants");
+
+/**
+ * Responsible for browser navigation based on given test locations.
+ * Each test location is an object which may contain:
+ * - name: name of this test
+ * - path: additional path to append to base url
+ * - viewport: viewport of the page (if omit, desktop view is used by default)
+ * - commands: a list of actions to perform on the page by the page navigator
+ */
+class PageOperator {
+  /**
+   * @param {PageNavigator} pageNavigator page navigator
+   * @param {import('puppeteer').Page} page A Pupeteer Page
+   * @param {Object[]} testLocations a list of test locations to navigate to
```

Comment:
Is there a type for `testLocation`? If not, that could be useful.

Reply:
Added more descriptive jsdoc for the object properties. No type since these test files are js only right now